pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1539,6 +1539,180 @@ func compareCacheWithNodeInfoSnapshot(cache *schedulerCache, snapshot *Snapshot)
 	return nil
 }
 
+func TestSchedulerCache_updateNodeInfoSnapshotList(t *testing.T) {
+	// Create a few nodes to be used in tests.
+	nodes := []*v1.Node{
+		// Node 0: a node with proper region and zone labels in a different region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-0",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-1",
+					v1.LabelZoneFailureDomain: "zone-1",
+				},
+			},
+		},
+		// Node 1: a node with proper region and zone labels in a different region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-1",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-1",
+					v1.LabelZoneFailureDomain: "zone-1",
+				},
+			},
+		},
+		// Node 2: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-2",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 3: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-3",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 4: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-4",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 5: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-5",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 6: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-6",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 7: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-7",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+	}
+
+	var cache *schedulerCache
+	var snapshot *Snapshot
+	type operation = func()
+
+	addNode := func(i int) operation {
+		return func() {
+			if err := cache.AddNode(nodes[i]); err != nil {
+				t.Error(err)
```

Comment:
nit: the t is a global (outer-most) testing instance, so a better practice is to pass in the sub-test `t` so that we can properly log the error message associated with the specific sub-test:

```
addNode := func(i int, t *testing.T)
```

(but just a nit as this line is less likely to fail)